components/button/src/Button.js

```diff
@@ -0,0 +1,304 @@
+/* @jsx jsx */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme } from '@westpac/core';
+import { mediaqueries } from './utils';
+
+// ==============================
+// Utils
+// ==============================
+
+// ==============================
+// Component
+// ==============================
+
+export const Button = ({
+	appearance,
+	size,
+	soft,
+	block,
+	trim,
+	icon: Icon,
+	iconPosition,
+	justify,
+	children,
+	tag: Tag,
+	onClick,
+	...props
+}) => {
+	const theme = useTheme();
+	const mq = mediaqueries(theme.breakpoints);
+
+	// Common styling
+	const styleCommon = {
+		textDecoration: 'none',
```

Comment:
Refactor styling as discussed (I've sent you my variations file on slack) :nail_care: